src/dos/drive_local.cpp

```diff
@@ -56,22 +56,43 @@ bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t /*attributes*
 	/* Test if file exists (so we need to truncate it). don't add to dirCache then */
 	bool existing_file = false;
 	
-	FILE * test = fopen_wrap(temp_name,"rb+");
+	FILE* test = fopen_wrap(temp_name, "rb");
 	if (test) {
 		fclose(test);
-		existing_file=true;
+		existing_file = true;
+	}
+
+	FILE* fp = nullptr;
 
+#ifdef WIN32
+	DWORD win_attribs = attributes ? (DWORD)attributes : FILE_ATTRIBUTE_NORMAL;
+	HANDLE win_fhandle = CreateFile(newname,
+	                                GENERIC_READ | GENERIC_WRITE,
+	                                0,
+	                                nullptr,
+	                                CREATE_ALWAYS,
+	                                win_attribs,
+	                                nullptr);
+	if (win_fhandle != INVALID_HANDLE_VALUE) {
+		int fd = _open_osfhandle((intptr_t)win_fhandle, _O_RDWR | _O_BINARY);
+		fp = _fdopen(fd, "wb+");
 	}
-	
-	FILE * hand = fopen_wrap(temp_name,"wb+");
-	if (!hand) {
+#else
+	int unix_perms = (attributes & DOS_ATTR_READ_ONLY) ? 0444 : 0666;
```

Comment:
PVS Analyzer shows warning V536 here even though it's not supposed to do that for multiple octal constants in close proximity. Don't know what's up with that.

Reply:
There are constants for this: https://www.gnu.org/software/libc/manual/html_node/Permission-Bits.html

But I'm not convinced omitting `fopen_wrap` here is safe - wouldn't it be better to continue using it, but to add a parameter specifying DOS attributes and handle them inside? AFAIK that's what the `cross.cpp` was created for.

Reply:
fopen_wrap is used in a whole bunch of places, not sure about moving all this stuff there. Plus it takes standard fopen mode parameter (rb, wb, etc) which would somehow need to be translated into custom parameters used by each platform.

Reply:
Yes, but it seems on some system `fopen_wrap` offers protection against accessing `/proc` filesystem - I'm not sure, why, but think this should better be preserved.
Moreover, since `cross.cpp` contains various filesystem abstraction routines, this code should go to `cross.cpp` too.

Reply:
Again, moving this into fopen_wrap is major scope change that would affect a lot more than this PR intends to cover. And there's still an issue of different open modes, translating them would be a headache.